parsec/remote_dep_mpi.c

```diff
@@ -1400,8 +1437,58 @@ static int remote_dep_mpi_pack_dep(int peer,
 #endif
     /* And now pack the updated message (msg->length and msg->output_mask) itself. */
     parsec_ce.pack(&parsec_ce, msg, dep_count, dep_dtt, packed_buffer, length, &saved_position);
+
+    /* If we're using the GET protocol, also pack the memory registration handles */
+    if (parsec_remote_dep_use_get) {
+        /* register memory and pack the memory handles */
+        for(k = 0; deps->outgoing_mask >> k; k++) {
+            if( !((1U << k) & deps->outgoing_mask )) continue;
+            if( !(deps->output[k].rank_bits[peer_bank] & peer_mask) ) continue;
+
+            void *dataptr = PARSEC_DATA_COPY_GET_PTR(deps->output[k].data.data);
+            MPI_Datatype dtt = deps->output[k].data.remote.dst_datatype;
+            int nbdtt = deps->output[k].data.remote.dst_count;
+
+            parsec_ce_mem_reg_handle_t source_memory_handle;
+            size_t source_memory_handle_size;
+
+            if(parsec_ce.capabilites.supports_noncontiguous_datatype) {
+                parsec_ce.mem_register(dataptr, PARSEC_MEM_TYPE_NONCONTIGUOUS,
+                                      nbdtt, dtt,
+                                      -1,
+                                      &source_memory_handle, &source_memory_handle_size);
+            } else {
+                /* TODO: Implement converter to pack and unpack */
+                int dtt_size;
+                parsec_type_size(dtt, &dtt_size);
+                parsec_ce.mem_register(dataptr, PARSEC_MEM_TYPE_CONTIGUOUS,
+                                      -1, NULL,
+                                      dtt_size,
```

Comment:
aren't we missing the count here ? parsec_type_size returns the size of the `dtt` type but it does not account for the `nbdtt`, so we need to scale it up for the mem_register in the contiguous case.

Reply:
Yes—this has long been fixed on my branch, see 948aa589e4b3854c38e99843c40bf5fdfb354cb6.

Reply:
@omor1 if you have a fix, would you mind upstreaming them to this branch?

Reply:
@omor1 any fixes that you have are more than welcomed.